### PR TITLE
API-5622 Replace customer::fields with session_fields.

### PR DESCRIPTION
**Pull request by gluszczyk**

### CODE REVIEW

File: content/messaging/agent-chat-api/changelog/index.mdx [hunk 1]
```diff
@@ -23,6 +23,7 @@ This document is the record of all the changes in the **Agent Chat API**, Web an
 - The [**Property**](/messaging/agent-chat-api/v3.2/#properties) data structure now has the same format in requests as in responses and pushes.
 - There's a new parameter `id` in the [Form field](/messaging/agent-chat-api/v3.2/#form-fields) data structure (`fields.answer.id` for the `group_chooser` field type).  It's an identifier for each option Customers can choose.
 - The **Add User to Chat** method ([Web](/messaging/agent-chat-api/v3.2/#add-user-to-chat) & [RTM](/messaging/agent-chat-api/v3.2/rtm-reference/#add-user-to-chat)) accepts an optional parameter `require_active_thread`.
+- Each method, response and push that deals with a variation of customer's fields had that field renamed to `session_fields` and its format changed. It's now an array that respects the item order.
```
Comment (ale-kacperczyk) [inline content/messaging/agent-chat-api/changelog/index.mdx]:
```suggestion
- Each method, response, and push that deals with a variation of customer `fields` had this field renamed to `session_fields` and its format changed. It's now an array that respects the item order.
```

File: content/messaging/agent-chat-api/v3.2/index.mdx [hunk 2]
```diff
@@ -891,19 +893,19 @@ Here's the list of all system messages you might come across:
...
-| `statistics`                     | optional  |                                           |
+| Field                            | Req./Opt. | Notes                                                                      |
+| -------------------------------- | --------- | -------------------------------------------------------------------------- |
+| `agent_last_event_created_at`    | optional  |                                                                            |
+| `avatar`                         | optional  |                                                                            |
+| `customer_last_event_created_at` | optional  |                                                                            |
+| `created_at`                     | optional  |                                                                            |
+| `email`                          | optional  |                                                                            |
+| `session_fields`                 | optional  | An array of custom object-enclosed `key:value`s. Expires with the session. |
```
Comment (ale-kacperczyk) [inline content/messaging/agent-chat-api/v3.2/index.mdx]:
```suggestion
| `session_fields`                 | optional  | An array of custom object-enclosed `key:value` pairs. Expires along with the session. |
```

File: content/messaging/agent-chat-api/v3.2/index.mdx [hunk 3]
```diff
@@ -3046,12 +3050,12 @@ Creates a new [Customer](#customer) user type.
...
-| `name`    | No       | `string`  |                                      |
-| `email`   | No       | `string`  |                                      |
-| `avatar`  | No       | `string`  | URL of the Customer's avatar         |
-| `fields`  | No       | `object`  | A map in the `"key": "value"` format |
+| Parameter        | Required | Data type  | Notes                                                                  |
+| ---------------- | -------- | ---------- | ---------------------------------------------------------------------- |
+| `name`           | No       | `string`   |                                                                        |
+| `email`          | No       | `string`   |                                                                        |
+| `avatar`         | No       | `string`   | URL of the Customer's avatar                                           |
+| `session_fields` | No       | `[]object` | An array of custom object-enclosed `key:value`s. Item order sensitive. |
```
Comment (ale-kacperczyk) [inline content/messaging/agent-chat-api/v3.2/index.mdx]:
```suggestion
| `session_fields` | No       | `[]object` | An array of custom object-enclosed `key:value` pairs. Respects the order of items. |
```

File: content/messaging/agent-chat-api/v3.2/index.mdx [hunk 4]
```diff
@@ -2827,18 +2829,18 @@ Returns the info about the Customer with a given `customer_id`.
...
+| `id`             | string    | [Customer's](#customer) ID.                                                                                                       |
+| `type`           | string    | `customer`                                                                                                                        |
+| `name`           | string    | Customer's name. Returned only if set.                                                                                            |
+| `email`          | string    | Customer's email. Returned only if set.                                                                                           |
+| `avatar`         | string    | Customer's avatar. Returned only if set.                                                                                          |
+| `created_at`     | string    | Specifies when the customer's identity was created.                                                                               |
+| `session_fields` | []object  | An array of custom object-enclosed `key:value`s. Returned only if set. Available for the duration of the session.                 |
```
Comment (ale-kacperczyk) [inline content/messaging/agent-chat-api/v3.2/index.mdx]:
```suggestion
| `session_fields` | []object  | An array of custom object-enclosed `key:value` pairs. Returned only if set. Available for the session duration.                 |
```

File: content/messaging/agent-chat-api/v3.2/index.mdx (hunk 2 above)
Comment (ale-kacperczyk) [reply content/messaging/agent-chat-api/v3.2/index.mdx]: I think `key:value` pairs looks better than `key:value`s

File: content/messaging/agent-chat-api/v3.2/index.mdx (hunk 3 above)
Comment (ale-kacperczyk) [reply content/messaging/agent-chat-api/v3.2/index.mdx]: Item order sensitive - this cluster is a bit confusing when you read it for the first time

File: content/messaging/customer-chat-api/v3.2/index.mdx [hunk 5]
```diff
@@ -2202,21 +2211,21 @@ curl -X POST \
...
+| **Webhook**            | [`customer_updated`](/management/configuration-api/#customer_updated)          |
 
 #### Request
 
-| Parameter | Required | Data type | Notes                              |
-| --------- | -------- | --------- | ---------------------------------- |
-| `fields`  | Yes      | `object`  | `key:value` format; max keys: 100  |
+| Parameter        | Required | Data type  | Notes                                                                                            |
+| ---------------- | -------- | ---------- | ------------------------------------------------------------------------------------------------ |
+| `session_fields` | No       | `[]object` | An array of custom object-enclosed `key:value` pairs. Respects the order of items. Max keys: 100 |
```
Comment (bkubiak) [inline content/messaging/customer-chat-api/v3.2/index.mdx]: not required?

Comment (bkubiak) [inline content/messaging/customer-chat-api/v3.2/index.mdx]: not required?

File: content/messaging/customer-chat-api/v3.2/rtm-reference/index.mdx [hunk 6]
```diff
@@ -2394,21 +2403,21 @@ Agent and referrer are updated by default using the browser’s headers.
...
+| **Push message**       | [`customer_updated`](#customer_updated)                          |
 
 #### Request
 
-| Parameter | Required | Data type | Notes                              |
-| --------- | -------- | --------- | ---------------------------------- |
-| `fields`  | Yes      | `object`  | `key:value` format; max keys: 100  |
+| Parameter        | Required | Data type  | Notes                                                                                            |
+| ---------------- | -------- | ---------- | ------------------------------------------------------------------------------------------------ |
+| `session_fields` | No       | `[]object` | An array of custom object-enclosed `key:value` pairs. Respects the order of items. Max keys: 100 |
```
Comment (bkubiak) [inline content/messaging/customer-chat-api/v3.2/rtm-reference/index.mdx]: not required?

File: content/messaging/customer-chat-api/v3.2/index.mdx (hunk 5 above)
Comment (bkubiak) [reply content/messaging/customer-chat-api/v3.2/index.mdx]: https://github.com/livechat/customer-api/blob/master/parsers/3.2/methods/set-customer-session-fields.go#L118

Comment (gluszczyk) [reply content/messaging/customer-chat-api/v3.2/index.mdx]: I see that this copy-paste error rattled you 😉

**Pull request merged**